### PR TITLE
Liquidation Visibility - Iteration One

### PR DESCRIPTION
**Pull request by anilhelvaci**
refs: #4 #3

### Security Considerations
No new authority is introduced to `vaultManager`. However, we make a new call to auctioneer inside `liquidateVaults` in order to fetch the `liveAuctionSchedule` which we expect to be fine. 

### Scaling Considerations

We introduced new storage nodes in `vstorage` which we don't plan to update once the data is written. 

### Testing Considerations
What tests should we implement other than the unit tests?

### Upgrade Considerations

These changes can be deployed via an upgrade to `vaultFactory`.

### CODE REVIEW

File: packages/inter-protocol/src/vaultFactory/vault.js [hunk 1]
```diff
@@ -131,6 +131,7 @@ export const VaultI = M.interface('Vault', {
   getCurrentDebt: M.call().returns(AmountShape),
   getNormalizedDebt: M.call().returns(AmountShape),
   getVaultSeat: M.call().returns(SeatShape),
+  getVaultState: M.call().returns(M.any()),
```
Comment (anilhelvaci) [inline packages/inter-protocol/src/vaultFactory/vault.js]: Might be good to introduce a type guard instead of `M.any()`

File: packages/inter-protocol/src/vaultFactory/vaultManager.js [hunk 2]
```diff
@@ -1197,10 +1276,31 @@ export const prepareVaultManagerKit = (
...
+              { ...data },
+            ],
+          );
+
           // This is expected to wait for the duration of the auction, which
           // is controlled by the auction parameters startFrequency, clockStep,
           // and the difference between startingRate and lowestRate.
-          const [proceeds] = await Promise.all([deposited, userSeatPromise]);
+          const [auctionSchedule, proceeds] = await Promise.all([
+            E(auctionPF).getSchedules(),
```
Comment (anilhelvaci) [inline packages/inter-protocol/src/vaultFactory/vaultManager.js]: At this point in time the auction is already finished. The question is whether or not we get the schedule where `liveScheduleAuction` is still pointing to the auction we just finished. This is what we need to indicate the `endTime` of the current liquidation auction.

I'm not sure if we can depend on `(E(acutionPF).getSchedule()).liveAuctionSchedule` is ___always___ going to point to right value due to distributed nature of the `vaultFactory` and `auctioneer` vats. Despite not being able put my finger directly on the possibility of a race condition, I still suggest we take the conservative path and call `E(acutionPF).getSchedule()` somewhere before await `deposited`.

File: packages/inter-protocol/src/vaultFactory/vaultManager.js [hunk 3]
```diff
@@ -1197,10 +1276,31 @@ export const prepareVaultManagerKit = (
               ),
           );
 
+          const [{ userSeatPromise, deposited }, liquidationRecorderKits] =
+            await Promise.all([
```
Comment (anilhelvaci) [inline packages/inter-protocol/src/vaultFactory/vaultManager.js]: Should we use `Promise.allSettled` instead of `Promise.all`? The reason being, if `helper.makeLiquidationRecorderKits` somehow fails `Promise.all` is going reject all together and we won't be able to complete the liquidation operation. What are the possible chances of `helper.makeLiquidationRecorderKits` failing, I don't know. However, when we think about it, `helper.makeLiquidationRecorderKits` interacts with another vat by calling `makeChildNode`. So depending on `chainStorage` vat to work correctly when settling liquidations seems like an unnecessary dependecy.

**Review by anilhelvaci — COMMENTED** [review]
What do you think? @Jorge-Lopes

File: packages/inter-protocol/src/vaultFactory/vault.js (hunk 1 above)
Comment (Jorge-Lopes) [reply packages/inter-protocol/src/vaultFactory/vault.js]: I agree.

File: packages/inter-protocol/src/vaultFactory/vaultManager.js (hunk 2 above)
Comment (Jorge-Lopes) [reply packages/inter-protocol/src/vaultFactory/vaultManager.js]: I agree.
In this case, we need to handle the promise rejection.
Some tests for this edge case will be required as well.

File: packages/inter-protocol/src/vaultFactory/vaultManager.js (hunk 3 above)
Comment (Jorge-Lopes) [reply packages/inter-protocol/src/vaultFactory/vaultManager.js]: I agree.
In this case, we need to handle the promise rejection.
Some tests for this edge case will be required as well.

**Review by Jorge-Lopes — APPROVED** [review]
The changes made in this PR implement the requested features and address the security concerns raised before.
